### ISSUE TITLE
Timepickers: Update to use Combobox

### ISSUE THREAD

**Issue by arodenbeck** [salesforce/design-system-react#1232]
Currently timepicker functions like a combobox, but screen readers only understand it to be an input.

**Comment by shleewhite** [comment]
@interactivellama if no one is working on this, I can do it.

**Comment by interactivellama** [comment]
No one is. Sounds great! It was started back in this issue. https://github.com/salesforce/design-system-react/pull/1430

The best thing may be to start from scratch and write a new component and just add some variants to Combobox. I'm not sure what that new component name would be though.

What would you like to propose?

**Comment by shleewhite** [comment]
I agree about just using the Combobox component... I'd be hesitant to change the name since it's called Timepicker in SLDS. Can I rewrite the component and keep the same name?
